core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -1108,6 +1108,24 @@ public Builder setDefaultPartitionSpec(int specId) {
       return this;
     }
 
+    Builder removeUnusedSpecs(Iterable<Integer> specIds) {
```

Comment:
to me it seems that `removeSpecIds` is slightly more descriptive in terms of naming, because the `Unused` part is rather misleading, since you're really just removing the specIds that were provided as input

Reply:
+1, in the context of a TableMetadata builder, I think the right name is `removeSpecs` or `removeSpecIds`. Whether it's unused or not is irrelevant for the purpose of the TableMetadata builder API